File: is_unique.cpp
#include <iostream>

using namespace std;

bool isUnique(string s) {
    if (s.length() > 128)  return false;

    bool characters[128] = {false};
    
    for(int i = 0; i < s.length(); i++) {
        if(characters[s[i]] == true) {
            return false;
        }
        else characters[(int)s[i]] = true;
    }
    return true;
}

int main() {
    cout << isUnique("Helo abc?") << endl;
    return 0;
}

File: src/array_and_strings/IsUnique.java
package array_and_strings;

public class IsUnique {
    public boolean isUnique(String s) {
        if(s.length() > 128) return false;

        boolean[] characters = new boolean[128];
        int index;

        for(int i = 0; i < s.length(); i++) {
            index = s.charAt(i);
            if(characters[index]){
                return false;
            }
            else characters[index] = true;
        }
        return true;
    }

    public static void main(String[] args) {
        IsUnique s = new IsUnique();
        System.out.println(s.isUnique("helo avc?"));
    }
}
